app/models/reviewable.rb

```diff
@@ -270,8 +270,15 @@ def log_history(reviewable_history_type, performed_by, edited: nil)
 
   def actions_for(guardian, args = nil)
     args ||= {}
+    built_actions =
+      Actions.new(self, guardian).tap { |actions| build_actions(actions, guardian, args) }
 
-    Actions.new(self, guardian).tap { |actions| build_actions(actions, guardian, args) }
+    # Empty bundles can cause big issues on the client side, so we remove them
+    # here. It's not valid anyway to have a bundle with no actions, but you can
+    # add a bundle via actions.add_bundle and then not add any actions to it.
+    built_actions.bundles.reject! { |bundle| bundle.actions.empty? }
+
+    built_actions
```

Comment:
Since `Bundle` is just a container with an array and metadata, I think we can implement at least `#empty?` on it, then we can do:

```suggestion
    built_actions.bundles.compact_blank
```

Reply:
Ooh nice, will do

Reply:
Ah the downside of that though is that I need to only affect the `bundles` array, I still need to return `built_actions` at the end. Still implemented `empty?`, just did this instead `built_actions.bundles.reject!(&:empty?)`